hardware_interface_testing/test/test_resource_manager.cpp

```diff
@@ -98,25 +98,29 @@ TEST_F(ResourceManagerTest, initialization_with_urdf)
 TEST_F(ResourceManagerTest, post_initialization_with_urdf)
 {
   TestableResourceManager rm;
-  ASSERT_NO_THROW(rm.load_urdf(ros2_control_test_assets::minimal_robot_urdf));
+  ASSERT_NO_THROW(rm.load_and_initialize_components(ros2_control_test_assets::minimal_robot_urdf));
 }
 
 TEST_F(ResourceManagerTest, test_unitilizable_hardware_validation)
 {
-  // If the the hardware can not be initialized and load_urdf tried to validate the interfaces a
-  // runtime exception is thrown
+  // If the the hardware can not be initialized and load_and_initialize_components tried to validate
+  // the interfaces a runtime exception is thrown
   TestableResourceManager rm;
-  ASSERT_THROW(
-    rm.load_urdf(ros2_control_test_assets::minimal_unitilizable_robot_urdf, true),
-    std::runtime_error);
+  EXPECT_FALSE(
+    rm.load_and_initialize_components(ros2_control_test_assets::minimal_unitilizable_robot_urdf));
 }
 
-TEST_F(ResourceManagerTest, test_unitilizable_hardware_no_validation)
+void test_load_and_initialized_components_failure(const std::string & urdf)
```

Comment:
While from a code duplication perspective this solution is obvious, it makes it rather hard to debug if the test actually fails. I would suggest adding some caller information to `test_load_and_initialized_components_failure` or using [`SCOPED_TRACE`](https://github.com/google/googletest/blob/main/docs/advanced.md#using-assertions-in-sub-routines) or [something similar](https://stackoverflow.com/a/53959810).

Reply:
I have added this to all relevant places :)

Thanks for the comment!